File: net/merge/Merger_pdf_xps.cs
namespace aspose_snippets.net
{
    public static partial class Merger
    {
        public static void pdf_to_xps()
        {
            var pathSource1 = @"..\..\TestData\test.pdf";
            var pathSource2 = @"..\..\TestData\Second\test.pdf";

            //read pdf file to Aspose Document
            var firstDoc = new Aspose.Pdf.Document(pathSource1);
            var secondDoc = new Aspose.Pdf.Document(pathSource2);

            //create empty pdf document
            var outputDoc = new Aspose.Pdf.Document();

            //set less memory usage with unload instead of fast performance
            outputDoc.EnableObjectUnload = true;

            foreach (var page in firstDoc.Pages)
            {
                //add page from one document to another directly
                outputDoc.Pages.Add(page);
            }

            foreach (var page in secondDoc.Pages)
            {
                //add page from one document to another directly
                outputDoc.Pages.Add(page);
            }

            //save xps document
            outputDoc.Save("test.xps",
                            new Aspose.Pdf.XpsSaveOptions
                            { 
                                //do not save transparent text to output file
                                SaveTransparentTexts = false
                            });
        }
    }
}

File: java/merge/src/aspose_snippets/java/Merger_pdf_xps.java
package aspose_snippets.java;

public class Merger_pdf_xps {
    public static void Execute() {
        String pathSource1 = "../../TestData/test.pdf";
        String pathSource2 = "../../TestData/Second/test.pdf";

        //read pdf file to Aspose Document
        com.aspose.pdf.Document firstDoc = new com.aspose.pdf.Document(pathSource1);
        com.aspose.pdf.Document secondDoc = new com.aspose.pdf.Document(pathSource2);

        //create empty pdf document
        com.aspose.pdf.Document outputDoc = new com.aspose.pdf.Document();

        //set less memory usage with unload instead of fast performance
        outputDoc.setEnableObjectUnload(true);

        for (com.aspose.pdf.Page page : firstDoc.getPages()) {
            //add page from one document to another directly
            outputDoc.getPages().add(page);
        }

        for (com.aspose.pdf.Page page : secondDoc.getPages()) {
            //add page from one document to another directly
            outputDoc.getPages().add(page);
        }

        com.aspose.pdf.XpsSaveOptions opt1 = new com.aspose.pdf.XpsSaveOptions();
        //do not save transparent text to output file
        opt1.setSaveTransparentTexts(false);
        //save xps document
        outputDoc.save("Merger_pdf_xps.xps", opt1);
    }
}
